docs/js/features/odata/use-generated-client.md

```diff
@@ -428,25 +428,99 @@ BusinessPartner.requestBuilder()
 
+An OData batch request can consist of a number of [retrieve requests](#retrieve-request) and [changesets](#changeset).
+
+#### Retrieve request
+
+A retrieve request is any HTTP `GET` request - in terms of the SAP Cloud SDK this includes all requests built by a [GetAllRequestBuilder](#getall-request-builder) and [GetByKeyRequestBuilder](#getbykey-request-builder).
+
+In the example below, a list of adresses is mapped in a [GetByKeyRequestBuilder](#getbykey-request-builder) to form the update requests, which are then passed to the batch function.
+
+The batch request will return a list of `Batchresponses`, which is stored in `retrieveResponses`.
+
+```ts
+async getAddresses(businessPartnerAddresses: BusinessPartnerAddress[])
+:Promise<BusinessPartnerAddress[]> {
+  const retrieveRequests = businessPartnerAddresses.map(
+      address => BusinessPartnerAddress
+        .requestBuilder()
+        .getByKey(address.businessPartner, address.addressId)
+  );
+
+  const [...retrieveResponses] = await batch(...retrieveRequests)
+    .execute(destination);
+      
+  return retrieveResponses.reduce(
+    (addresses, response: ReadResponse) =>
+      [...addresses, ...response.as(BusinessPartnerAddress)], []
+  );
+}
+```
+
+#### Changeset
+
+A changeset is a collection of HTTP `POST`, `PUT`, `PATCH` and `DELETE` operations - requests built by any [CreateRequestBuilder](#create-request-builder), [UpdateRequestBuilder](#update-request-builder) and [DeleteRequestBuilder](#delete-request-builder) in terms of the SAP Cloud SDK. The order of execution within a changeset is not defined as opposed to the whole batch request itself. Therefore the requests within a changeset should not depend on each other. If the execution of any of the requests within a changeset fails, the whole changeset will be reflected as an error in the response and will not be applied, much like a database transaction.
+
+In the example below, a list of adresses is mapped in an [UpdateRequestBuilder](#update-request-builder) to form the update requests, which are then wrapped in a changeset and passed to the batch function.
+
+The batch request will return a single `Batchresponse` in a list, which is stored in `updateChangesetResponse`.
+
 ```ts
-async updateAddreses(businessPartnerAddresses: BusinessPartnerAddress[]): Promise<BusinessPartnerAddress[]> {
+updateAddresses(businessPartnerAddresses: BusinessPartnerAddress[])
+:Promise<BusinessPartnerAddress[]> {
   const updateRequests = businessPartnerAddresses.map(
-    address => BusinessPartnerAddress.requestBuilder().update(address)
+    address => BusinessPartnerAddress
+      .requestBuilder()
+      .update(address)
   );
 
+  const [updateChangesetResponse] = await batch(changeset(...updateRequests))
+    .execute(destination);
+
+  return (updateChangesetResponse as WriteResponses)
+    .responses
+    .map(
+      response => response.as(BusinessPartnerAddress)
+    );
+}
```

Comment:
What I miss in this section is the usefulness of the Responses, what information can I get from it (e. g. response status code).